packages/backend/src/open_payments/wallet_address/middleware.ts

```diff
@@ -10,20 +10,56 @@ export function createWalletAddressMiddleware() {
     ctx: AppContext,
     next: () => Promise<unknown>
   ): Promise<void> => {
-    if (
-      ctx.path === '/incoming-payments' ||
-      ctx.path === '/outgoing-payments' ||
-      ctx.path === '/quotes'
-    ) {
-      if (ctx.method === 'GET') {
-        ctx.walletAddressUrl = ctx.query['wallet-address'] as string
-      } else if (ctx.method === 'POST') {
-        ctx.walletAddressUrl = (ctx.request.body as CreateBody).walletAddress
+    if (ctx.method === 'GET') {
```

Comment:
We should break this up into multiple middlewares or at least separate functions in this file

Reply:
Later. Please create issue 😉